native/wasmex/src/module.rs

```diff
@@ -139,8 +139,17 @@ fn val_type_to_atom(val_type: &ValType) -> Atom {
         ValType::F32 => atoms::f32(),
         ValType::F64 => atoms::f64(),
         ValType::V128 => atoms::v128(),
-        ValType::ExternRef => atoms::extern_ref(),
-        ValType::FuncRef => atoms::func_ref(),
+        ValType::Ref(_ref_type) => atoms::extern_ref(),
+    }
+}
+
+fn ref_type_to_atom(ref_type: &RefType) -> Atom {
+    let heap_type = ref_type.heap_type();
+
+    if heap_type.is_func() {
+        atoms::func_ref()
+    } else {
+        atoms::extern_ref()
```

Comment:
I'm not entirely sure if this is sufficient or correct. Please advise.

Reply:
Perhaps `(ref null any)` is broader and a better fit? With a new atom, something like `:any_null_ref` or `:ref_null_any`?

Relevant wasmtime documentation: https://docs.wasmtime.dev/api/wasmtime/struct.RefType.html

Reply:
hmm, looking at [the `Display` trait implementation for `HeapType`](https://docs.wasmtime.dev/api/src/wasmtime/runtime/types.rs.html#689) (and [same for `RefType`](https://docs.wasmtime.dev/api/src/wasmtime/runtime/types.rs.html#363)) I wonder whether we should make this return a binary with a concrete type string instead of just saying fun/extern ref. Since (as far as I understand) we can have any type here, right? And debugging some wasm error might be MUCH easier if we see what exact type we have instead of just `extern_ref`.

what do you folks think? can we just return `format!("{ref_type}")` as a binary and it's even better and easier? Am I missing something? Downside is that we expect atoms here - but that can be changed 🤔

Reply:
Ok. I can do that.

Reply:
As requested [32d1abd](https://github.com/tessi/wasmex/pull/599/commits/32d1abdbb4ebaed024d19a399fcb27623dc500dd)